src/api/app/controllers/webui/projects/bs_requests_controller.rb

```diff
@@ -26,34 +25,23 @@ def index
 
       private
 
-      def filter_by_involvement(involvement)
-        return filter_by_involvement_staging_project(involvement) if staging_projects.present?
+      def set_bs_requests
+        return unless Flipper.enabled?(:request_index, User.session)
 
-        case involvement
-        when 'all'
-          target = BsRequest.with_actions_and_reviews.where(bs_request_actions: { target_project: @project.name })
-          source = BsRequest.with_actions_and_reviews.where(bs_request_actions: { source_project: @project.name })
-          reviews = BsRequest.with_actions_and_reviews.where(reviews: { project: @project, package: nil })
-          target.or(source).or(reviews)
-        when 'incoming'
-          BsRequest.with_actions.where(bs_request_actions: { target_project: @project.name })
-        when 'outgoing'
-          BsRequest.with_actions.where(bs_request_actions: { source_project: @project.name })
-        end
+        @bs_requests = @project.bs_requests
       end
 
-      def filter_by_involvement_staging_project(involvement)
-        case involvement
-        when 'all'
-          target = BsRequest.with_actions_and_reviews.where(staging_project: staging_projects, bs_request_actions: { target_project: @project.name })
-          source = BsRequest.with_actions_and_reviews.where(staging_project: staging_projects, bs_request_actions: { source_project: @project.name })
-          reviews = BsRequest.with_actions_and_reviews.where(staging_project: staging_projects, reviews: { project: @project, package: nil })
-          target.or(source).or(reviews)
-        when 'incoming'
-          BsRequest.with_actions.where(staging_project: staging_projects, bs_request_actions: { target_project: @project.name })
-        when 'outgoing'
-          BsRequest.with_actions.where(staging_project: staging_projects, bs_request_actions: { source_project: @project.name })
-        end
+      def filter_involvement
+        return if params[:involvement]&.compact_blank.blank?
+
+        @selected_filter['involvement'] = params[:involvement]
+        @bs_requests =  case
+                        when @selected_filter['involvement'].include?('incoming')
+                          @bs_requests.where(bs_request_actions: { target_project_id: @project.id })
+                                      .or(@bs_requests.where(reviews: { by_project: @project.name }))
```

Comment:
And also here, you could apply your fix of "switching away from strings to user_id/project_id/package_id"...

Reply:
done